cesium/features/lomb_scargle.py

```diff
@@ -84,22 +109,75 @@ def lomb_scargle_model(
                 nharm=nharm,
                 detrend_order=0,
             )
-        model_dict["freq_fits"].append(fit)
-        signal -= fit["model"]
-        model_dict["freq_fits"][-1]["resid"] = signal.copy()
+
+        # remove current estim_freq for next run [**NORM]
+        norm_residual = signal - fit["model"]
+        signal = norm_residual
+
+        # store current estim_freq results, [**RESCALED for first fund freq]
+        current_fit = (
+            rescale_lomb_model(fit, mmean, mscale)
+            if (opt_normalize & (i == 0))
+            else fit
+        )
+        model_dict["freq_fits"].append(current_fit)
+        model_dict["freq_fits"][-1]["resid"] = norm_residual
+
+        if opt_normalize & (i == 0):
```

Comment:
These two if clauses feel as if they should be combined higher up, but perhaps that's tricky.